index.src.html

```diff
@@ -830,18 +870,28 @@ <h4 id="algorithm-create" algorithm>Create a `Credential`</h4>
             types in order to support a "sign-up" use case. For the moment, though, we're punting
             on that by restricting the dictionary to a single entry.
 
-    5.  Let |p| be [=a new promise=].
+    6.  Let |p| be [=a new promise=].
 
-    6.  Run the following steps [=in parallel=]:
+    7.  Let |origin| be |settings|'s [=environment settings object/origin=].
 
-        1.  Let |r| be the result of executing |interfaces|[0] {{Credential/[[Create]](options)}}
-            internal method on |options|.
+    8.  Run the following substeps [=in parallel=]:
 
-        2.  If |r| is an [=exception=], [=reject=] |p| with |r|.
+        1.  Let |r| be the result of executing |interfaces|[0]'s {{Credential/[[Create]](origin, options)}}
+            internal method given |origin| and |options|.
 
-            Otherwise, [=resolve=] |p| with |r|.
+        2.  If |r| is an [=exception=] or `null`, [=reject=] |p| with |r|, and terminate these substeps.
 
-    7.  Return |p|.
+        3.  If |r| is a {{Credential}}, [=resolve=] |p| with |r|, and terminate these substeps.
+
+        4.  Assert: |r| is a <code>[=constructCredential=]</code> algorithm.
+
+        5.  [=Queue a task=] on |global|'s [=DOM manipulation task source=] to run the following substeps:
+
+            1. Let |result| be the result of [=promise-calling=] |r| given |global|.
+            2. If |result| is an [=exception=], [=reject=] |p| with |result| and terminate these substeps.
```

Comment:
Once you've promise-called |r|, you can just [=resolve=] |p| with the result, since promise-calling turns exceptions into rejected promises, and resolving a promise with a rejected promise rejects the outer one.

Reply:
doh, ok.